biosys/apps/main/api/views.py

```diff
@@ -272,8 +273,64 @@ def get_serializer_context(self):
                 ctx['species_mapping'] = self.species_facade_class().name_id_by_species_name()
         return ctx
 
+    def search_data(self, qs, search_param, order_by=None):
+        """
+        Search does not support searching within JSONField.
+        :param qs: queryset
+        :param search_param: value to search
+        :param order_by: field to order by, prefixed with '-' for descending order
+        :return: the queryset after search filters applied
+        """
+        field_names = [field['name'] for field in self.dataset.data_package['resources'][0]['schema']['fields']]
```

Comment:
The recommended way to get the field names:
```
field_name = self.dataset.schema.field_names
```

Reply:
Noted, will change